android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Interactor.kt

```diff
@@ -150,26 +150,15 @@ public abstract class Interactor<P : Any, R : Router<*>>() : InteractorType, Rib
   }
 
   internal fun setRouterInternal(router: Router<*>) {
-    if (routerDelegate != null) {
-      this.router = router as R
-    }
+    this.router = router as R
   }
 
   /** @return the currently attached presenter if there is one */
   @VisibleForTesting
   private fun getPresenter(): P {
-    val presenter: P? =
-      try {
-        if (actualPresenter != null) {
-          actualPresenter
-        } else {
-          injectedPresenter
-        }
-      } catch (e: UninitializedPropertyAccessException) {
-        actualPresenter
-      }
-    checkNotNull(presenter) { "Attempting to get interactor's presenter before being set." }
-    return presenter
+    return runCatching { actualPresenter ?: injectedPresenter }
```

Comment:
`runCatching` is designed for wrapping some exception for later rethrowing/handling. It should almost never be used as a catch-all that just swallows the generic `Throwable` (it catches everything). 

I suggest we maintain catching only the specific exception, though the code for sure could be improved in style somehow.

See [Encapsulate successful or failed function execution](https://github.com/Kotlin/KEEP/blob/master/proposals/stdlib/result.md) for use cases and motivation of `runCatching`